tcms/testcases/templates/testcases/mutable.html

```diff
@@ -81,7 +81,7 @@
                 <div class="col-md-3 col-lg-3 {% if form.priority.errors %}has-error{% endif %}">
                     <select id="id_priority" name="priority" class="form-control selectpicker">
                         {% for priority in form.priority.field.queryset %}
-                            <option value="{{ priority.pk }}" {% if priority.pk == form.priority.pk %}selected{% endif %}>
+                            <option value="{{ priority.pk }}" {% if priority.pk == test_case.priority.pk %}selected{% endif %}>
```

Comment:
This is breaking error handling. When you have errors in some other field, e.g. unknown default tester the page will be reloaded and the values for other fields must remain whatever has been selected.

When EditCaseForm is initialized in the view I see that we pass the field ID but for some reason even if the change is reverted the value isn't remembered. This is a bug in the template which also happens for New TestCase page so you will have to fix that first.

Reply:
The change here shouldn't be necessary if the edit form is initialized properly with all the field values and the template is checking what it is supposed to check.

Reply:
Found the problem. But we still need a change. The correct check is `{% if priority.pk == form.priority.value %}`

Reply:
and more - `priority.pk` is an `int` while `form.priority.value` is `string`, so I had to apply the `strinformat:"d"` pipe on the former. And we had the same bug on `status` because the logic there is the same, so I fixed it on both places

Reply:
:+1:

Reply:
for the record Product/Category also seem to not be properly selected when the error is rendered after an error handling (e.g. non existing user). I think these widgets need to be fixed first before we can continue with the rest.

There's the same problem for new TestPlan page as well!

Reply:
For `Product` the problem and fix were the same. We were comparing `int` with `string`, so adding `stringformat` solved the issue.
For `Category` is a bit tricky - adding `stringformat` solves the problem.. at first. However, after the page is loaded we execute the javascript and reloads the categories, selecting the first one. We must come up with a workaround. I can think of two:
1. on page load we check whether there is something in the category dropdown. If yes, we don't call `populateProductCategory` since we know we have the categories we need.
2. when calling `populateProductCategory` we pass an `id` to be selected after reload. By default will be 0, but for such cases we may pass the id of the category that comes preselected.
IMO Plan 1 sound better, what do you think?

Reply:
Anyhow, I implemented 1) because it was easy enough. If you are ok with this, we can merge it, and I will proceed with fixing these issues in TestPlan page. If not, any better ideas will be welcomed.